torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -912,6 +939,9 @@ def _post_backward_final_callback(
     root_state = state
 
     if root_state._sync_gradients:
+        # TODO: also waits for optimizer step to finish. This can be pushed to the next forward
+        # by recording an event for each individually wrapped FSDP module, and only waiting
```

Comment:
In order for this mentioned solution to work, do we need to run the overlapped optimizer outside the post-backward stream and in a separate stream? Otherwise, `wait_stream()` is too coarse grained to differentiate between the optimizer computation and other existing post-backward ops.